air/src/air/logup_gkr.rs

```diff
@@ -81,9 +82,72 @@ pub trait LogUpGkrEvaluator: Clone + Sync {
     }
 }
 
+#[derive(Clone, Default)]
+pub struct DummyLogUpGkrEval<B: StarkField, P: Clone + Send + Sync + ToElements<B>> {
+    _field: PhantomData<B>,
+    _public_inputs: PhantomData<P>,
+}
+
+impl<B, P> LogUpGkrEvaluator for DummyLogUpGkrEval<B, P>
+where
+    B: StarkField,
+    P: Clone + Send + Sync + ToElements<B>,
+{
+    type BaseField = B;
+
+    type PublicInputs = P;
+
+    fn get_oracles(&self) -> &[LogUpGkrOracle<Self::BaseField>] {
+        panic!("LogUpGkrEvaluator method called but LogUp-GKR is not implemented")
+    }
+
+    fn get_num_rand_values(&self) -> usize {
+        panic!("LogUpGkrEvaluator method called but LogUp-GKR is not implemented")
+    }
+
+    fn get_num_fractions(&self) -> usize {
+        panic!("LogUpGkrEvaluator method called but LogUp-GKR is not implemented")
+    }
+
+    fn max_degree(&self) -> usize {
+        panic!("LogUpGkrEvaluator method called but LogUp-GKR is not implemented")
+    }
+
+    fn build_query<E>(&self, _frame: &EvaluationFrame<E>, _periodic_values: &[E], _query: &mut [E])
+    where
+        E: FieldElement<BaseField = Self::BaseField>,
+    {
+        panic!("LogUpGkrEvaluator method called but LogUp-GKR is not implemented")
+    }
+
+    fn evaluate_query<F, E>(
+        &self,
+        _query: &[F],
+        _rand_values: &[E],
+        _numerator: &mut [E],
+        _denominator: &mut [E],
+    ) where
+        F: FieldElement<BaseField = Self::BaseField>,
+        E: FieldElement<BaseField = Self::BaseField> + ExtensionOf<F>,
+    {
+        panic!("LogUpGkrEvaluator method called but LogUp-GKR is not implemented")
+    }
+
+    fn compute_claim<E>(&self, _inputs: &Self::PublicInputs, _rand_values: &[E]) -> E
+    where
+        E: FieldElement<BaseField = Self::BaseField>,
+    {
+        panic!("LogUpGkrEvaluator method called but LogUp-GKR is not implemented")
+    }
+}
+
 #[derive(Clone, Debug, PartialEq, PartialOrd, Eq, Ord)]
 pub enum LogUpGkrOracle<B: StarkField> {
+    // a column with a given index in the main trace segment
     CurrentRow(usize),
+    // a column with a given index in the main trace segment but shifted upwards
     NextRow(usize),
+    // a virtual periodic column defined by its values in a given cycle. Note that the cycle length
+    // must be a power of 2.
     PeriodicValue(Vec<B>),
```

Comment:
nit: we should use doc comments here - i.e., `///`.

Reply:
Changed